includes/Core/REST_API/REST_Routes.php

```diff
@@ -281,6 +343,59 @@ function ( $response ) {
 					),
 				)
 			),
+			new REST_Route(
+				'core/user/data/user-input-settings',
+				array(
+					array(
+						'permission_callback' => '__return_true', // It gets checked in the User_Input_Settings class.
```

Comment:
Even though we do some checks in the callback, I think we should still leverage this rather than making it public. We should at least require `Permissions::AUTHENTICATE` as all users submitting settings need to be able to connect. We require more than that internally but I think that should be a minimum for this endpoint to be active. Then the comment that it is checked in the other class is unnecessary.

Also, would you please move the key to be after `callback`? Not a big deal but this is how others are defined.

Reply:
Updated.